src/mesh/BoundingBox.cpp

```diff
@@ -13,50 +13,63 @@ namespace precice::mesh {
 
 logging::Logger BoundingBox::_log{"mesh::BoundingBox"};
 
+BoundingBox::BoundingBox(Eigen::VectorXd boundMin, Eigen::VectorXd boundMax)
+{
+  // PRECICE_ASSERT 1
+  PRECICE_ASSERT(boundMin.rows() == boundMax.rows(), "Dimension of min ({}) and max ({}) vertices should be the same.", boundMin.rows(), boundMax.rows());
+
+  // PRECICE_ASSERT 2
+  PRECICE_ASSERT((boundMin - boundMax).maxCoeff() < 0, "Each component of min vertex ({}) must be <= max vertex ({}) in the same axis direction.", boundMin, boundMax);
+
+  _boundMin   = std::move(boundMin);
+  _boundMax   = std::move(boundMax);
+  _dimensions = _boundMin.rows();
+}
+
 BoundingBox::BoundingBox(std::vector<double> bounds)
 {
-  PRECICE_ASSERT((int) bounds.size() == 4 || (int) bounds.size() == 6, "Dimension of a bounding box can only be 2 or 3.", bounds.size() / 2);
-  _bounds     = std::move(bounds);
-  _dimensions = _bounds.size() / 2;
+  PRECICE_ASSERT((int) bounds.size() == 4 || (int) bounds.size() == 6, "Dimension of a bounding box can only be 2 or 3. The dimension is ({}).", bounds.size() / 2);
+  _dimensions = bounds.size() / 2;
+  _boundMin   = Eigen::Map<Eigen::VectorXd, 0, Eigen::InnerStride<2>>(bounds.data(), _dimensions);
+  _boundMax   = Eigen::Map<Eigen::VectorXd, 0, Eigen::InnerStride<2>>(bounds.data() + 1, _dimensions);
 }
 
 BoundingBox::BoundingBox(int dimension)
     : _dimensions(dimension)
 {
-  PRECICE_ASSERT(dimension == 2 || dimension == 3, "Dimension of a bounding box can only be 2 or 3.", dimension);
-  for (int i = 0; i < _dimensions; ++i) {
-    _bounds.push_back(std::numeric_limits<double>::max());
-    _bounds.push_back(std::numeric_limits<double>::lowest());
-  }
+  PRECICE_ASSERT(dimension == 2 || dimension == 3, "Dimension of a bounding box ({}) can only be 2 or 3.", dimension);
+
+  _boundMin = Eigen::VectorXd(dimension);
+  _boundMax = Eigen::VectorXd(dimension);
+
+  // Define 'illegal' BoundingBox: _boundMin > _boundMax
+  std::fill(_boundMin.data(), _boundMin.data() + _boundMin.rows(), std::numeric_limits<double>::max());
+  std::fill(_boundMax.data(), _boundMax.data() + _boundMax.rows(), std::numeric_limits<double>::lowest());
 }
 
 bool BoundingBox::operator==(const BoundingBox &otherBB) const
 {
-  PRECICE_ASSERT(_dimensions == otherBB._dimensions, "Bounding boxes with different dimensions cannot be compared.", _dimensions, otherBB._dimensions);
-  for (int i = 0; i < _dimensions; ++i) {
-    if (_bounds.at(i) != otherBB._bounds.at(i)) {
-      return false;
-    }
-  }
-  return true;
+  PRECICE_ASSERT(_dimensions == otherBB._dimensions, "Bounding boxes ({} and {}) with different dimensions cannot be compared.", _dimensions, otherBB._dimensions);
+
+  return _boundMin.isApprox(otherBB._boundMin) && _boundMax.isApprox(otherBB._boundMax);
```

Comment:
In case anyone is interested in checking: https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c